### PR TITLE
Popover: use a11y hooks instead of HoCs

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Replaces the constrained tabbing and focus return HoCs with the new hooks.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by youknowriad** [comment]
haha, duplicate of #27675

**Comment by ellatrix** [comment]
@youknowriad Oh, I didn't see. I'm happy you're trying this too. I don't think it can be done without wrapping the merged ref callback in `useCallback`.

**Comment by ellatrix** [comment]
Closing in favour of #27675

**Comment by youknowriad** [comment]
yes, what do yo think of the idea here https://github.com/WordPress/gutenberg/pull/27710 still against it?

**Comment by ellatrix** [comment]
Let's try and see if it works :) All that matters here is Popover using the hooks

File: packages/components/src/popover/index.js [hunk 1]
```diff
@@ -417,8 +418,17 @@ const Popover = ( {
...
+	const focusReturnRef = useFocusReturn();
 	const focusOnMountRef = useFocusOnMount( focusOnMount );
 	const focusOutsideProps = useFocusOutside( handleOnFocusOutside );
+	const allRefs = [
+		containerRef,
+		focusOnMount ? constrainedTabbingRef : null,
+		focusOnMount ? focusReturnRef : null,
+		focusOnMountRef,
+	];
+	const mergedRefs = useCallback( mergeRefs( allRefs ), allRefs );
```
Comment (ellatrix) [inline packages/components/src/popover/index.js:431]: `useCallback` is necessary here, I'm getting e2e test failures otherwise.

**Pull request merged**